File: agorachat/src/main/java/io/agora/chatdemo/base/BaseInitActivity.java
package io.agora.chatdemo.base;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;

public abstract class BaseInitActivity extends BaseActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        int layoutId = getLayoutId();
        if(layoutId != 0) {
            setContentView(layoutId);
        }else {
            setContentView(getContentView());
        }
        initSystemFit();
        initIntent(getIntent());
        initView(savedInstanceState);
        initListener();
        initData();
    }

    protected View getContentView() {
        return null;
    }

    /**
     * get layout id
     * @return
     */
    protected int getLayoutId() {
        return 0;
    }

    protected void initSystemFit() {
        setFitSystemForTheme(true);
    }


    /**
     * init intent
     * @param intent
     */
    protected void initIntent(Intent intent) { }

    /**
     * init view
     * @param savedInstanceState
     */
    protected void initView(Bundle savedInstanceState) {

    }

    /**
     * init listener
     */
    protected void initListener() { }

    /**
     * init data
     */
    protected void initData() { }

}

File: app-kotlin/src/main/kotlin/io/agora/chatdemo/base/BaseInitActivity.kt
package io.agora.chatdemo.base

import android.content.Intent
import android.os.Bundle
import androidx.viewbinding.ViewBinding
import io.agora.uikit.base.EaseBaseActivity

abstract class BaseInitActivity<B : ViewBinding> : EaseBaseActivity<B>() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initIntent(intent)
        initView(savedInstanceState)
        initListener()
        initData()
    }

    /**
     * init intent
     * @param intent
     */
    protected open fun initIntent(intent: Intent?) {}

    /**
     * init view
     * @param savedInstanceState
     */
    protected open fun initView(savedInstanceState: Bundle?) {}

    /**
     * init listener
     */
    protected open fun initListener() {}

    /**
     * init data
     */
    protected open fun initData() {}
}